### PR TITLE
abort(404) when retrieving invalid journal

### PR DESCRIPTION
**Pull request by amdomanska**
Fixes: https://github.com/DOAJ/doajPM/issues/2129

### CODE REVIEW

**Comment by amdomanska** [comment]
I added the check if the identifier is an id (len = 32) if not ISSN. I suggest postponing adding new regex checks for now as that could require further and more in-depth research.

File: portality/view/doaj.py [hunk 1]
```diff
@@ -253,20 +253,27 @@ def toc(identifier=None, volume=None, issue=None):
     # identifier may be the journal id or an issn
     journal = None
     issn_ref = False
+
     if len(identifier) == 9:
```
Comment (Steven-Eardley) [inline portality/view/doaj.py]: We have the opportunity here to fix another bug - `None` (the default value for `identifier`) will cause an exception if you give it to `len()`. This means there's an error when you go to https://doaj.org/toc/

File: portality/view/doaj.py [hunk 2]
```diff
@@ -253,20 +253,27 @@ def toc(identifier=None, volume=None, issue=None):
...
-    else:
-        journal = models.Journal.pull(identifier)  # Returns None on fail
+        issn_ref = True  # just a flag so we can check if we were requested via issn
+    elif len(identifier) == 32:
+        js = models.Journal.pull(identifier)  # Returns None on fail
 
-    if journal is None:
-        abort(404)
+        if js is None:
+            abort(400)
```
Comment (amdomanska) [inline portality/view/doaj.py]: it should be `abort(404)`

File: portality/view/doaj.py [hunk 3]
```diff
@@ -253,20 +253,27 @@ def toc(identifier=None, volume=None, issue=None):
...
+        journal = js
+    else:
+        abort(400)
```
Comment (amdomanska) [inline portality/view/doaj.py]: as above

**Pull request closed without merging**